### PR TITLE
Update checkmarx.yml attributes : "uses","project","teams","scanners","params".

### PR DESCRIPTION
**Pull request by DhavalPatelPersistent**
## Pre-requisites

- [ ] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [ ] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [ ] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [ ] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [ ] Should include comments in the workflow for any parts that are not obvious or could use clarification.

**For _CI_ workflows, the workflow:**

- [ ] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [ ] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [ ] Packaging workflows should run on `release` with `types: [ created ]`.
- [ ] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [ ] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [ ] `name`: Name of the Code Scanning integration.
  - [ ] `organization`: Name of the organization producing the Code Scanning integration.
  - [ ] `description`: Short description of the Code Scanning integration.
  - [ ] `categories`: Array of languages supported by the Code Scanning integration.
  - [ ] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [ ] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [ ] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [ ] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [ ] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [ ] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

File: code-scanning/checkmarx.yml [hunk 1]
```diff
@@ -29,14 +29,16 @@ jobs:
     - uses: actions/checkout@v2
     # Runs the Checkmarx Scan leveraging the latest version of CxFlow - REFER to Action README for list of inputs
     - name: Checkmarx CxFlow Action
-      uses: checkmarx-ts/checkmarx-cxflow-github-action@04e6403dbbfee0fd3fb076e5791202c31c54fe6b
+      uses: checkmarx-ts/checkmarx-cxflow-github-action@master
```
Comment (anaarmas) [inline code-scanning/checkmarx.yml:32]: Using a branch ref instead of a concrete sha goes against our guidelines here

**Review by anaarmas — CHANGES_REQUESTED** [review]
@DhavalPatelPersistent would you mind providing some context for these changes?

**Comment by DhavalPatelPersistent** [comment]
Hello Ana,

Cx-Flow is the actual Product that is generation SARIF report but  configuration/parameter for Cx-flow are in default checkmarx.yml.

On setting up workflow default configuration, which is fetch form this file checkmarx.yml is not correct/latest and so SARIF report has issues.

Though we can update *.yml file in workflow folder afterward, it will be more convenient to fetch defaul workflow/*.yml with latest configuration.

Below is the URL which is fetching this file to setup the workflow:
https://github.com/DhavalPatelPersistent/hello-world-java/new/master?filename=.github%2Fworkflows%2Fcheckmarx-analysis.yml&workflow_template=checkmarx&code_scanning

Thanks & Regards,
Dhaval Patel
[https://github.githubassets.com/images/modules/open_graph/github-logo.png]<https://github.com/DhavalPatelPersistent/hello-world-java/new/master?filename=.github%2Fworkflows%2Fcheckmarx-analysis.yml&workflow_template=checkmarx&code_scanning>
Build software better, together<https://github.com/DhavalPatelPersistent/hello-world-java/new/master?filename=.github%2Fworkflows%2Fcheckmarx-analysis.yml&workflow_template=checkmarx&code_scanning>
GitHub is where people build software. More than 73 million people use GitHub to discover, fork, and contribute to over 200 million projects.
github.com



________________________________

**Comment by anaarmas** [comment]
Thanks @DhavalPatelPersistent! We will still need a concrete SHA rather than a branch name for the action reference before merging this, though - whatever is the latest commit in master should be fine.

**Comment by DhavalPatelPersistent** [comment]
Hello Ana, Latest SHA for "checkmarx-cxflow-github-action" is 9975af7d6b957abec9ee9646effa3fb3b82c5314.

**Review by anaarmas — APPROVED** [review]
Wonderful, thanks @DhavalPatelPersistent!

**Pull request merged**